common/djangoapps/microsite_configuration/microsite.py

```diff
@@ -98,13 +121,23 @@ def get_value_for_org(org, val_name, default=None):
 
 def get_all_orgs():
     """
-    This returns a set of orgs that are considered within a microsite. This can be used,
+    This returns a set of orgs that are considered within all microsites. This can be used,
     for example, to do filtering
     """
     org_filter_set = set()
     if not has_configuration_set():
         return org_filter_set
 
+    if "microsite_configuration.middleware.DatabaseMicrositeMiddleware" in settings.MIDDLEWARE_CLASSES:
+        # Get the orgs in the db
+        candidates = Microsite.objects.all()
+        for microsite in candidates:
+            current = json.loads(microsite.values)
+            org_filter = current.get('course_org_filter')
+            if org_filter:
+                org_filter_set.add(org_filter)
+
+    # Get the orgs in the settings file
```

Comment:
I like that. That would mean there is a django.setting that selects the microsite backend and these get_all_orgs method, and posible get_value_for_org only call the backend method, right?

I think these methods would be required:
- has_configuration_set
- get_template_path (posibly)
- get_value_for_org
- get_all_orgs
- _set_current_microsite
- set_by_domain

Then we could remove set_from_db_by_domain, and _set_current_microsite_from_obj

Do you think that would make this, the sort of PR that changes too much and then it is too dificult to review?